internal/yggdrasil/configuration-assembler.go

```diff
@@ -100,15 +110,32 @@ func (a *configurationAssembler) getDeviceConfiguration(ctx context.Context, edg
 	return &dc, nil
 }
 
-func getHeartbeatConfiguration(edgeDevice *v1alpha1.EdgeDevice) *models.HeartbeatConfiguration {
-	if edgeDevice.Spec.Heartbeat != nil {
+func getOsConfiguration(edgeDevice *v1alpha1.EdgeDevice, deviceGroup *v1alpha1.EdgeDeviceGroup) *models.OsInformation {
+	osInformation := edgeDevice.Spec.OsInformation
```

Comment:
this means no patch-merge of group's configuration by device's configuration. 
for simplicity, we can start with it. but we might consider changing it in the future.

Reply:
Yes - that's a complete override. That was the intention - start small and modify when needed (supported by user feedback).